packages/flutter_tools/lib/src/migrate/migrate_compute.dart

```diff
@@ -0,0 +1,625 @@
+          platforms: platforms,
+        );
+        if (newDirectoryPath != context.migrateResult.generatedBaseTemplateDirectory?.path) {
+          context.migrateResult.generatedBaseTemplateDirectory = context.fileSystem.directory(newDirectoryPath);
+        }
+        // Determine merge type for each newly generated file.
+        final List<FileSystemEntity> generatedBaseFiles = context.migrateResult.generatedBaseTemplateDirectory!.listSync(recursive: true);
+        for (final FileSystemEntity entity in generatedBaseFiles) {
+          if (entity is! File) {
+            continue;
+          }
+          final File baseTemplateFile = entity.absolute;
+          final String localPath = getLocalPath(baseTemplateFile.path, context.migrateResult.generatedBaseTemplateDirectory!.absolute.path, context.fileSystem);
+          if (!context.migrateResult.mergeTypeMap.containsKey(localPath)) {
+            // Use two way merge when the base revision is the same as the target revision.
+            context.migrateResult.mergeTypeMap[localPath] = revision == targetRevision ? MergeType.twoWay : MergeType.threeWay;
+          }
+        }
+        if (newDirectoryPath != context.migrateResult.generatedBaseTemplateDirectory?.path) {
+          context.migrateResult.generatedBaseTemplateDirectory = context.fileSystem.directory(newDirectoryPath);
+          break; // The create command is old and does not distinguish between platforms so it only needs to be called once.
+        }
+      }
+    }
+  }
+}
+
+class MigrateTargetFlutterProject extends MigrateFlutterProject {
+  MigrateTargetFlutterProject({
+    required super.path,
+    required super.directory,
+    required super.name,
+    required super.androidLanguage,
+    required super.iosLanguage,
+    super.platformWhitelist,
+  });
+
+  /// Creates the base reference app based off of the migrate config in the .metadata file.
+  Future<void> createProject(
+    MigrateContext context,
+    String targetRevision,
+    Directory targetFlutterDirectory,
+  ) async {
+    if (path == null) {
+      // Create target
+      context.status.pause();
+      context.logger.printStatus('Creating target app with revision $targetRevision.', indent: 2, color: TerminalColor.grey);
+      context.status.resume();
+      if (context.verbose) {
+        context.logger.printStatus('Creating target app.');
+      }
+      await context.migrateUtils.createFromTemplates(
+        targetFlutterDirectory.childDirectory('bin').absolute.path,
+        name: name,
+        androidLanguage: androidLanguage,
+        iosLanguage: iosLanguage,
+        outputDirectory: context.migrateResult.generatedTargetTemplateDirectory!.absolute.path,
+      );
+    }
+  }
+}
+
+/// Parses the metadata of the flutter project, extracts, computes, and stores the
+/// revisions that the migration should use to migrate between.
+class MigrateRevisions {
```

Comment:
This is a strange class, I guess it's mutating the context, but on initialization? That's unexpected.